From d4c0d4cdcc0001480c32f56465844b8f239eb1ec Mon Sep 17 00:00:00 2001
From: Andreas Blaesius <skate4life@gmx.de>
Date: Wed, 18 Jan 2017 06:29:28 +0100
Subject: [PATCH 19/21] Add back FORCE_SCREENSHOT_CPU_PATH CPP_FLAG

Change-Id: I68ca3e633a2903c68a67ffbebdc26a6541a5f621
---
 BoardConfigCommon.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 784b8c1..4073f9c 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -43,6 +43,10 @@ TARGET_SPECIFIC_HEADER_PATH += $(OMAP4_NEXT_FOLDER)/domx/omx_core/inc
 BOARD_USE_TI_CUSTOM_DOMX := true
 DOMX_PATH := $(OMAP4_NEXT_FOLDER)/domx
 
+ifdef OMAP_ENHANCEMENT
+    TARGET_RELEASE_CPPFLAGS += -DFORCE_SCREENSHOT_CPU_PATH
+endif
+
 # Bootanimation
 TARGET_BOOTANIMATION_PRELOAD := false
 TARGET_BOOTANIMATION_TEXTURE_CACHE := false
-- 
2.7.4

